convert2rhel/toolopts.py

```diff
@@ -97,40 +129,59 @@ def _get_argparser():
             "  convert2rhel [--no-rhsm] [--disablerepo repoid]"
             " [--enablerepo repoid] [--no-rpm-va] [--debug] [--restart] [-y]\n"
             "  convert2rhel [-k activation_key | -c conf_file_path] [-o organization] [--pool pool_id | -a] [--disablerepo repoid] [--enablerepo"
-            " repoid] [--serverurl url] [--keep-rhsm] [--no-rpm-va] [--debug] [--restart] [-y]"
+            " repoid] [--serverurl url] [--keep-rhsm] [--no-rpm-va] [--debug] [--restart] [-y]\n"
+            r"  convert2rhel {analyze}"
             "\n\n"
-            "*WARNING* The tool needs to be run under the root user"
+            "*WARNING* The tool needs to be run under the root user\n"
         )
         return argparse.ArgumentParser(
-            prog="convert2rhel",
-            conflict_handler="resolve",
-            usage=usage,
-            add_help=False,
+            conflict_handler="resolve", usage=usage, formatter_class=argparse.RawTextHelpFormatter
         )
 
-    def _register_options(self):
-        """Prescribe what command line options the tool accepts."""
-        self._parser.add_argument(
-            "-h",
-            "--help",
-            action="help",
-            help="Show help message and exit.",
+    def _register_commands(self):
+        """Configures parsers specific to the analyze and convert subcommands"""
+        subparsers = self._parser.add_subparsers(title="Subcommands", dest="command")
+        self._analyze_parser = subparsers.add_parser(
+            "analyze",
+            help="Run all Convert2RHEL initial checks up until the "
+            " Point of no Return (PONR) and generate a report with the findings."
+            " A rollback is initiated after the checks to put the system back"
+            " in the original state.",
+            parents=[self._shared_options_parser],
+            formatter_class=argparse.RawTextHelpFormatter,
+            usage=SUPPRESS,
         )
-        self._parser.add_argument(
+        self._convert_parser = subparsers.add_parser(
+            "convert",
+            help="Convert the system.",
+            parents=[self._shared_options_parser],
+            formatter_class=argparse.RawTextHelpFormatter,
+            usage=SUPPRESS,
```

Comment:
See my comment to fix this: https://github.com/oamg/convert2rhel/pull/820#issuecomment-1686462026